packages/backend-tasks/src/tasks/TaskWorker.ts

```diff
@@ -220,19 +253,22 @@ export class TaskWorker {
 
   async tryReleaseTask(
     ticket: string,
-    settings: TaskSettingsV1,
+    settings: TaskSettingsV2,
   ): Promise<boolean> {
-    const { recurringAtMostEveryDuration } = settings;
+    const isCron = !settings?.cadence.startsWith('P');
+
+    const n = settings as TaskSettingsV2;
+    const time = isCron
+      ? new CronTime(n.cadence).sendAt().toISOString()
+      : DateTime.now().plus(Duration.fromISO(n.cadence)).toISO();
```

Comment:
hm ok but here it actually gets worse - it used to be that it computed the previous `next_run_start_at` + a delta, NOT now + delta. I think for the non-cron case you actually should try to preserve the old behavior.